libs/os/os_test.go

```diff
@@ -135,3 +135,44 @@ func newTestProgram(t *testing.T, environVar string) (cmd *exec.Cmd, stdout *byt
 
 	return
 }
+
+// Ensure that using CopyFile does not truncate the destination file before
+// the origin is positively a non-directory and that it is ready for copying.
+// See https://github.com/tendermint/tendermint/issues/6427
+func TestTrickedTruncation(t *testing.T) {
+	tmpDir, err := ioutil.TempDir(os.TempDir(), "pwn_truncate")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer os.Remove(tmpDir)
```

Comment:
It's a bit of a nit but I think you can just do this right
```suggestion
	tmpDir := t.TempDir()
```

Reply:
Thanks @cwgoes for the change point out to use Go1.15's (*testing.T).TempDir(), but I intentionally didn't use it for maximum compatibility. Sure we want to only make this build on Go1.15 and above?

Reply:
So I know Tendermint has a minimum requirement of Go1.15 and this is just a test so I don't think it's too much of a problem, but I also don't mind if you leave it as is

Reply:
We should also backport this

Reply:
No worries. Yeah that sounds good